oak_functions/loader/src/server.rs

```diff
@@ -490,88 +509,50 @@ fn check_export_function_signature(
 /// if allowed by the policy. If the execution of the function takes longer than allowed by the
 /// policy, or the resulting response is larger than allowed by the policy, an error response
 /// generated by the policy is returned instead of the actual response.
-async fn apply_policy<F, S>(policy: Option<Policy>, function: F) -> anyhow::Result<Response<Body>>
+async fn apply_policy<F, S>(
+    policy: Option<Policy>,
+    function: F,
+) -> anyhow::Result<http::response::Response<Body>>
 where
     F: std::marker::Send + 'static + FnOnce() -> S,
-    S: std::future::Future<Output = anyhow::Result<FunctionsResponse>> + std::marker::Send,
+    S: std::future::Future<Output = anyhow::Result<Response>> + std::marker::Send,
 {
     match policy {
-        None => Response::<Body>::try_from(function().await?).context("couldn't get the response"),
+        None => create_http_response(function().await?),
         Some(policy) => {
             // Use tokio::spawn to actually run the tasks in parallel, for more accurate measurement
             // of time.
             let task = tokio::spawn(async move { function().await });
             tokio::time::sleep(policy.constant_processing_time).await;
 
-            let response = task.now_or_never();
+            let function_response = task.now_or_never();
 
-            match response {
-                None => policy.create_error_response("response not available"),
+            let response = match function_response {
+                None => Response {
+                    status: StatusCode::PolicyTimeViolation as i32,
+                    body: "Reason: response not available".as_bytes().to_vec(),
```

Comment:
Ideally we should be able to tell these two cases apart, if we manage to get tokio parallelism to work! If the function terminated without a response, but still terminated before the policy timeout, then `function_response` would not be `None`. It would contain an error, which is handled in the other match arm.